### PR TITLE
Remove duplicates from JarList

### LINKED ISSUE #34357: Fix repetitive JAR loading in Ballerina running and testing phases
**Description:**
It seems when we have the same JAR in two different stdlibs, Ballerina running and testing phases repetitively load both the JARs and due to that there could be JAR resolution errors like [#2492](https://github.com/ballerina-platform/ballerina-standard-library/issues/2492).

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Remove duplicates from JarList for building and testing

Fixes #34357 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java [hunk 1]
```diff
@@ -131,8 +131,14 @@ private void addPlatformLibraryPaths(PackageContext packageContext,
         // Add all the jar library dependencies of current package (packageId)
         Collection<PlatformLibrary> otherJarDependencies = jBalBackend.platformLibraryDependencies(
                 packageContext.packageId(), scope);
+
+        List<String> fileNames = new ArrayList();
+        libraryPaths.stream().forEach(e -> fileNames.add(e.path().toFile().getName()));
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java:136]: AFAIU, this filter will not work as intended. The requirement is to stop adding a jar if a path with the same filename already exists in the jarList.

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java:136]: This seems to be correct. Lets test and verify

**Review by azinneera — APPROVED** [review]

**Comment by Ibaqu** [comment]
We tested the fix at 2 states in GRPC : at the latest commit and as well as the commit where the issue was first found (95de4d) 

For the latest commit, the netty changes were reverted (https://github.com/BuddhiWathsala/module-ballerina-grpc/commit/124c0e54bad05e76853e970b8eedec00ecbed0ed) and the fix was tested with on the latest stage branch. It should be noted that the ballerina-parser version 2.0.0-beta.6 was to be used to make sure the tests ran. 

For the commit (95de4d), the HTTP module was using an old tcnative-boringssl jar which had to be updated for the tests to run.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java [hunk 2]
```diff
@@ -131,8 +131,14 @@ private void addPlatformLibraryPaths(PackageContext packageContext,
         // Add all the jar library dependencies of current package (packageId)
         Collection<PlatformLibrary> otherJarDependencies = jBalBackend.platformLibraryDependencies(
                 packageContext.packageId(), scope);
+
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java:134]: IMO, the better solution would be to make a `jarFiles` a `Set` and override the `equals` method of `JarLibrary`

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java:134]: @sameerajayasoma is this ok to be addressed in a PR later? RC1 is waiting on this.

Comment (azinneera) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java:134]: @sameerajayasoma would it be fine to address this in a PR later. RC1 is waiting on this PR.

**Pull request merged**